src/Gif.ts

```diff
@@ -1,26 +1,20 @@
 /**
- * A DataObject that represents a Gif.
+ * A Data Object that represents a Gif.
  *
  * @property originalUrl - The URL of the full size GIF
  * @property thumbnailUrl - The URL of the thumbnail GIF
  */
 export class Gif {
+    constructor(
+        readonly originalUrl: string,
+        readonly thumbnailUrl: string
+    ) {}
+}
 
-    private readonly _originalUrl: string;
-    private readonly _thumbnailUrl: string;
-
-    constructor(originalUrl: string, thumbnailUrl: string) {
-        this._originalUrl = originalUrl;
-        this._thumbnailUrl = thumbnailUrl;
-    }
-
-    /** @returns The URL for the full size Gif file */
-    get originalUrl() {
-        return this._originalUrl;
-    }
-
-    /** @returns The URL for the thumbnail size Gif file */
-    get thumbnailUrl() {
-        return this._thumbnailUrl;
-    }
+/**
+ * Type Guard to determine if an object is a Gif
+ * @param nilable can be an `any` or `nil` (`null|undefined`)
+ */
+export const isGif = (nilable:any|null|undefined): nilable is Gif => {
```

Comment:
I would use the `unknown` type for this as it would fit the context and usage.

```suggestion
export const isGif = (nilable: unknown): nilable is Gif => {
```

Reply:
I always forget about `unknown`. Good idea!